src/Maestro/Maestro.Web/Controllers/DependenciesController.cs

```diff
@@ -0,0 +1,44 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.HealthMetrics;
+using Microsoft.Extensions.Logging;
+
+namespace Maestro.Web.Controllers
+{
+    [Route("[controller]")]
+    [Route("_/[controller]")]
+    public class DependenciesController : ControllerBase
+    {
+        private readonly IRemoteFactory remoteFactory;
+        private readonly ILogger<DependenciesController> logger;
+
+        public DependenciesController(IRemoteFactory factory, ILogger<DependenciesController> logger)
+        {
+            remoteFactory = factory;
+            this.logger = logger;
+        }
+
+        [HttpGet("getSubscriptionDependencyDetails/{host}/{account}/{project}/{*branch}")]
+        public async Task<IActionResult> GetSubscriptionDependencyDetails(string host, string account, string project, string branch)
+        {
+            string repo = "";
+
+            if (host == "github")
+            {
+                repo = $"https://github.com/{account}/{project}";
+            }
+            else
+            {
+                repo = $"https://dev.azure.com/{account}/internal/_git/{project}";
```

Comment:
We need to factor this behavior out. It's leaking all over the place in our code, and they've already changed it once.  Honestly, I just wish we didn't depend on this at all (we really, really shouldn't).  Why can't we just let the user pass us the whole string so we aren't depending on the format here?

Reply:
For the same reason as I pulled the full string from the file request controller in the old version - we're going to auth whatever string they send us, so we should be the ones constructing the string. If you don't feel like the same security issue applies here then I can pass the whole string, but we should definitely do a security analysis on it.

Reply:
The only "security" feature this has is that is hard codes "internal"... but that's sort of... anti-security... since that's the private one. The only alternative is "public" which doesn't have an code in it.  I'd just take the string and save on all the logic.

Reply:
I guess it also ensures we are at least talking to GitHub or Azure DevOps... mostly I'd just like this logic to get isolated somewhere.  We parse and construct these URL's in quite a few places, so if it's ever changes again, we're going to be hunting for the for a while.  Though just taking the string and ensuring it's host is either github.com or dev.azure.com would manage that too (but should probably still be factored out)